pep-0677.rst

```diff
@@ -0,0 +1,854 @@
+expressible using Callback Protocols, and since they are uncommon we
+decided that it made more sense to move forward with a simpler syntax.
+
+Extended Syntax Supporting Named and Optional Arguments
+-------------------------------------------------------
+
+Another alternative was for a compatible but more complex syntax that
+could express everything in this PEP but also named, optional, and
+variadic arguments. In this “extended” syntax proposal the following
+types would have been equivalent::
+
+    class Function(typing.Protocol):
+        def f(self, x: int, /, y: float, *, z: bool = ..., **kwargs: str) -> bool:
+            ...
+
+    Function = (int, y: float, *, z: bool = ..., **kwargs: str) -> bool
+
+Advantages of this syntax include: - Most of the advantages of the
+proposal in this PEP (conciseness, PEP 612 support, etc) -
+Furthermore, the ability to handle named, optional, and variadic
+arguments
+
+We decided against proposing it for the following reasons:
+
+- The implementation would have been more difficult, and usage stats
+  demonstrate that fewer than 3% of use cases would benefit from any
+  of the added features.
+- The group that debated these proposals was split down the middle
+  about whether these changes are even desirable:
+
+  - On the one hand, they make callable types more expressive. On the
+    other hand, they could easily confuse users who have not read the
+    full specification of callable type syntax.
+  - We believe the simpler syntax proposed in this PEP, which
+    introduces no new semantics and closely mimics syntax in other
+    popular languages like Kotlin, Scala, and TypesScript, is much
+    less likely to confuse users.
+
+- We intend to implement the current proposal in a way that is
+  forward-compatible with the more complicated extended syntax. If the
+  community decides after more experience and discussion that we want
+  the additional features, they should be straightforward to propose
+  in the future.
+- We realized that because of overloads, it is not possible to replace
+  all need for Callback Protocols even with an extended syntax. This
+  makes us prefer proposing a simple solution that handles most use
+  cases well.
+
+We confirmed that the current proposal is forward-compatible with
+extended syntax by
+`implementing <https://github.com/stroxler/cpython/tree/callable-type-syntax--extended>`_
+a quick-and-dirty grammar and AST on top of the grammar and AST for
+the current proposal.
+
+
+Syntax Closer to Function Signatures
+------------------------------------
+
+One alternative we had floated was a syntax much more similar to
+function signatures.
+
+In this proposal, the following types would have been equivalent::
+
+    class Function(typing.Protocol):
+        def f(self, x: int, /, y: float, *, z: bool = ..., **kwargs: str) -> bool:
+            ...
+
+    Function = (x: int, /, y: float, *, z: bool = ..., **kwargs: str) -> bool
```

Comment:
Same substitutions in the previous example: `s/Function/MyCallback/g` and `s/f/__call__`